apps/system/lockscreen/js/lockscreen.js

```diff
@@ -518,10 +530,17 @@
 
   LockScreen.prototype.setPassCodeEnabled =
   function ls_setPassCodeEnabled(val) {
+    var value;
     if (typeof val === 'string') {
-      this.passCodeEnabled = val == 'false' ? false : true;
+      value = val == 'false' ? false : true;
     } else {
-      this.passCodeEnabled = val;
+      value = val;
+    }
+
+    if (this.passCodeEnabled instanceof LockScreen.Deferred) {
+      this.passCodeEnabled.resolve(value);
+    } else {
+      this.passCodeEnabled = value;
```

Comment:
This line should be moved out of the `else` block right to move `this.passCodeEnabled` back into a boolean?